rest/test_results_routes.go

```diff
@@ -65,6 +65,59 @@ func (h *testResultsGetByTaskIdHandler) Run(ctx context.Context) gimlet.Responde
 	return gimlet.NewJSONResponse(testResults)
 }
 
+///////////////////////////////////////////////////////////////////////////////
+//
+// GET /test_results/display_task_id/{display_task_id}
+
+type testResultsGetByDisplayTaskIDHandler struct {
+	opts data.TestResultsOptions
+	sc   data.Connector
+}
+
+func makeGetTestResultsByDisplayTaskID(sc data.Connector) gimlet.RouteHandler {
+	return &testResultsGetByDisplayTaskIDHandler{
+		sc: sc,
+	}
+}
+
+// Factory returns a pointer to a new testResultsGetByDisplayTaskIDHandler.
+func (h *testResultsGetByDisplayTaskIDHandler) Factory() gimlet.RouteHandler {
+	return &testResultsGetByDisplayTaskIDHandler{
+		sc: h.sc,
+	}
+}
+
+// Parse fetches the task_id from the http request.
+func (h *testResultsGetByDisplayTaskIDHandler) Parse(_ context.Context, r *http.Request) error {
+	var err error
+
+	h.opts.DisplayTaskID = gimlet.GetVars(r)["display_task_id"]
+	vals := r.URL.Query()
+	if len(vals[execution]) > 0 {
+		h.opts.Execution, err = strconv.Atoi(vals[execution][0])
+		return err
+	} else {
```

Comment:
nit: since you already return in the if clause, you don't need to have else here.